### DISCUSSION TITLE
What is the best way to send HTTP requests in sequence in one react component?

### DISCUSSION THREAD

**Discussion by nshen** [urql-graphql/urql#3284] [Q&A] [answered]
Hi, I want to send several requests in react component, and each request depends on the result of the previous one.
and when I refetch the first one, I want all requests in sequences to execute one by one. This is very common but I didn't find how to do it in documents, does anyone have an example of it?

**Comment by kitten** [comment, marked as answer]
You can use `pause` to ensure you're not fetching until you have all inputs (i.e. variables) available. Subsequently, changing variables will trigger refetches automatically.

However, please DON'T DO THIS.
This totally negates the point of having a GraphQL API and modelling it to a) serve your UI structure, b) model relationships in data, c) declaratively define your data dependencies. 😅

**Comment by nshen** [comment]
hi @kitten  do you know why I got  too many re-render error with the code below :

```tsx
export const useList = () => {

  const [{ data: audiosCountData, }, refetch] = useGetAudiosCountQuery()
  const audiosCount = audiosCountData?.aggregateAudio._max?.id
  const ids = audiosCount ? getRandomIds(audiosCount as number, 20) : null
  const [{
    data,
    error,
    fetching,
  }] = useFindManyAudioQuery({
    pause: !ids,
    variables: {
      where: {
        id: {
          in: ids
        }
      }
    }
  })

  console.log(data, fetching)
  return [data, error, fetching, refetch]
}


```

**Reply by kitten** [reply]
Well, are your variables being randomised on every render 😅
